File: src/ios/Sources/AnsightToolsVisualTree/QueryOverlaysTool.swift
import AnsightCore
import Foundation

public final class QueryOverlaysTool: AnsightTool {
    public init() {}

    public var descriptor: AnsightToolDescriptor {
        AnsightToolDescriptor(
            id: AnsightVisualTreeToolIds.queryOverlays,
            name: "Query Overlays",
            description: "Lists live diagnostic overlays and supports simple metadata filtering.",
            category: "ui",
            scope: AnsightToolScope.read.rawValue,
            keywords: "ui overlay highlight query list metadata",
            security: AnsightVisualTreeToolSecurityProfiles.queryOverlays,
            argumentsSchema: AnsightVisualTreeToolSchemas.queryOverlaysArguments,
            resultSchema: AnsightVisualTreeToolSchemas.queryOverlaysResult
        )
    }

    public func execute(arguments: [String: String]) throws -> AnsightToolExecutionResult {
        AnsightVisualTreeOverlaySupport.queryOverlays(arguments: arguments)
    }
}

File: src/dotnet/Ansight.Tools.VisualTree/QueryOverlaysTool.cs
namespace Ansight.Tools.VisualTree;

public sealed class QueryOverlaysTool : ITool
{
    public string Category => "ui";

    public ToolPolicy Policy => ToolPolicy.Read;

    public string Id => VisualTreeToolIds.QueryOverlays;

    public string Name => "Query Overlays";

    public string Description => "Lists live diagnostic overlays and supports simple metadata filtering.";

    public string Keywords => "ui overlay highlight query list metadata";

    public ToolSchema ArgumentsSchema => VisualTreeToolSchemas.QueryOverlaysArguments;

    public ToolSchema ResultSchema => VisualTreeToolSchemas.QueryOverlaysResult;

    public Task<ToolResult> Execute(IReadOnlyDictionary<string, string> arguments)
    {
        ArgumentNullException.ThrowIfNull(arguments);
        return VisualTreeSupport.QueryOverlaysAsync(arguments);
    }
}
